poc/groups.sage

```diff
@@ -40,6 +40,40 @@ def OS2IP_le(octets, skip_assert=False):
         assert octets == I2OSP_le(ret, len(octets))
     return ret
 
+class ScalarField:
+
+    def __init__(self, order):
+        self.field = GF(order)  # Delegate field operations to GF instance
```

Comment:
Eventually I gave up the inheritance approach due to the complexity.
GF is not a simple class. It is created using a [UniqueFactory](https://doc.sagemath.org/html/en/reference//structure/sage/structure/factory.html#sage.structure.factory.UniqueFactory). To initialize it properly, we need __new__ instead of __init__, which causing a series of new complexity.